DogTales/dog/player.cpp

```diff
@@ -27,6 +23,12 @@ void Player::tick(bave::Seconds const dt) {
 
 void Player::draw(bave::Shader& shader) const { m_sprite.draw(shader); }
 
+float const Player::get_controller_state(std::string key) {
+	try {
+		return m_player_controller.key_map[key];
+	} catch (std::out_of_range const& oor) { return -1.f; }
```

Comment:
We don't want to risk exceptional paths in hot code like this.

Reply:
what should i do instead? just return the element? or use `if()`?

Reply:
First of all the member function should be `const`. Once that's done `key_map[key]` will not compile anyway. So you'd want to use `key_map.find(key)` and return a default value / nullopt if the key isn't present.